### PR TITLE
Feat/add connection endpoint for promoting draft connection

### PR DESCRIPTION
**Pull request by maxchehab**
Creates a `sso.createConnection({ source })` method. Should be used in replacement of `sso.promoteDraftConnection({ token })`.

A depreciation message was added to `sso.promoteDraftConnection`. We should remove the method completely at the next major version.

### CODE REVIEW

File: src/sso/sso.spec.ts [hunk 1]
```diff
@@ -154,5 +154,45 @@ describe('SSO', () => {
         }
       });
     });
+
+    describe('createConnection', () => {
+      it('send valid token', async () => {
```
Comment (marktran) [inline src/sso/sso.spec.ts:159]:
```
describe('with valid token', () => {
  it('responds with a 201 CREATED status', () => {
  })
})

describe('with invalid token', () => {
  it('responds with a 404 NOT FOUND status', () => {
  })
})
```

File: src/sso/sso.ts [hunk 2]
```diff
@@ -46,6 +55,11 @@ export class SSO {
   }
 
   async promoteDraftConnection({ token }: PromoteDraftConnectionOptions) {
+    //  tslint:disable:no-console
+    console.error(
```
Comment (marktran) [inline src/sso/sso.ts:59]: Can we add a logger instead of using console.*

**Comment by maxchehab** [comment]
Added an optional `logger: stream.Transform` which defaults to a Winston instance. Developers can provide their own loggers as an optional parameter.

Update `createConnection` and `promoteDraftConnection` tests.

File: src/sso/sso.ts (hunk 2 above)
Comment (marktran) [reply src/sso/sso.ts:59]: How about `console.warn`? Seems more appropriate for a deprecation.

**Pull request merged**